### DISCUSSION TITLE
Start nagging users to upgrade from Windows 7

### DISCUSSION THREAD

**Discussion by qaisjp** [multitheftauto/mtasa-blue#1205] [Ideas]
Windows 7 will no longer receive security updates from 17 January 2020. This is Friday next week.

We should nag players to upgrade to Windows 10. We can also provide links [like this](https://www.cnet.com/how-to/upgrade-to-windows-10-free-heres-how/) that explain how to upgrade to Windows 10 for free (or add our own instructions to the wiki).

**Why so soon?**

It's a precautionary measure.

Our support for XP went on for far too long. Mainstream support for XP ended on April 14, 2009; and extended support ended on April 8, 2014. We only stopped supporting XP in July-ish 2019 — and we still only stop releasing builds for XP in our next minor version (1.6).

Announcing this now gives our players plenty of notice and will hopefully shift our player numbers more towards Windows 10.

**Are there any dependencies that don't support Windows 7?**

Not that I'm aware of.

Google Chrome will [still be supported until "at least July 15, 2021"](https://cloud.google.com/blog/products/chrome-enterprise/how-chrome-is-helping-enterprises-still-using-windows-7). Google Chrome isn't CEF but I can't find details for CEF support, so this will do as an estimate. CEF will probably support Windows 7 for longer, but who knows.

**Current statistics (updated Dec 29, 2020)**

- **Windows 10**: 71.9%
- **Windows 8.1**: 4.8%
- **Windows 8**: 0.6%
- **Windows 7**: 22.7%
- **Windows Vista / XP**: <0.04% (about only 50 people!)

**Comment by ffsPLASMA** [comment]
Are there any technical reasons why you want to get rid of win7? From what I understood, you had to maintain 2 branches for XP/Vista and 7/8/10. Does this sort of apply here aswell? I doubt those players will upgrade soon, also MS announcing extended security updates until 2023.

**Comment by MegaThorx** [comment]
> I doubt those players will upgrade soon, also MS announcing extended security updates until 2023.

The extended support isn't free [1].

> With that in mind, today we are announcing that we will offer paid Windows 7 Extended Security Updates (ESU) through January 2023. The Windows 7 ESU will be sold on a per-device basis and the price will increase each year. 

And regular ends on January 14, 2020 [2]

[1] https://www.microsoft.com/en-us/microsoft-365/blog/2018/09/06/helping-customers-shift-to-a-modern-desktop/
[2] https://support.microsoft.com/en-us/help/13853/windows-lifecycle-fact-sheet

**Comment by qaisjp** [comment]
> Are there any technical reasons why you want to get rid of win7? From what I understood, you had to maintain 2 branches for XP/Vista and 7/8/10. Does this sort of apply here aswell? I doubt those players will upgrade soon, also MS announcing extended security updates until 2023.

There are none right now, so this is not a proposal to get rid of Windows 7 support (because we don't need to... yet).

Indeed, we had to maintain two branches. [The XP branch contained](https://github.com/multitheftauto/mtasa-blue/compare/release/xp-vista):
- a number of XP specific fixes
- frozen CEF (master uses newer versions of CEF)

Maintaining two separate long-running branches is a pain (a pain thankfully taken on by ccw) and it would be ideal to not have to do it again.

There are no technical changes yet so it doesn't apply here right now, we can still just use `master` (things aren't going to break overnight).

**Comment by Vadya963** [comment]
if you switch to Windows 10, then only after the MTA is 64-bit, because everyone is sitting on a 64-bit system, the 21st century, and the MTA is still 32-bit

**Comment by ffsPLASMA** [comment]
> 
> 
> if you switch to Windows 10, then only after the MTA is 64-bit, because everyone is sitting on a 64-bit system, the 21st century, and the MTA is still 32-bit

MTA will never be 64Bit unless somehow MTA team gets source code of San Andreas^^

**Comment by patrikjuvonen** [comment]
What are the latest OS usage statistics for MTA?

**Comment by patrikjuvonen** [comment]
Updated OS usage statistics:

Windows 10: 71.9% (up from 58.6%)
Windows 8.1: 4.8% (down from 6.3%)
Windows 8: 0.6% (down from 0.9%)
Windows 7: 22.7% (down from 34.22%)
_Deprecated: Windows Vista / XP: <0.04% (about only 50 people!) (down from 0.2%)_

**Comment by patrikjuvonen** [comment]
Windows 10+: 88.0%>
Windows 8.1: <2.6%
Windows 8: <0.2%
Windows 7: <9.2%
_Deprecated: Windows Vista / XP: <0.1% (specifically 6 users)_
_Other: <0.1%_
